x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Block.java

```diff
@@ -121,6 +121,7 @@ public interface Block extends Accountable, BlockLoader.Block, NamedWriteable, R
     enum MvOrdering {
         UNORDERED(false, false),
         DEDUPLICATED_UNORDERD(true, false),
+        SORTED_ASCENDING(false, true),
```

Comment:
We are serializing the MvOrdering as well and a newer ESQL version reading from an old version Enum I _think_ it will be confused by the position in the Enum of this new value. Take a look at `StreamOutput.writeEnum`.

What I'm trying to say is that, because the enum values are serialized using their values' position in the enum, new values added to a serializable enum should be added at the end.

Reply:
In the rest of Elasticsearch we implement enum serialization by hand rather than by ordinal. Well, we've mostly been doing that. I think SQL forked from a time when we were doing it by ordinal.

We transitioned because of cases like this - the "right" way to solve this is to send the key for `UNORDERED` to older node versions on the wire. If the serialization code was explicit then you'd be able to do it right in the enum itself.

Reply:
I'm fine pulling this out of the fix if that makes things easier to land. We can get it in a followup.

Reply:
All good, that's easy to get right - just missed it initially :)